plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/DefaultValidationJob.java

```diff
@@ -18,44 +18,29 @@
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.OperationCanceledException;
 import org.eclipse.core.runtime.Status;
-import org.eclipse.core.runtime.jobs.Job;
 import org.eclipse.emf.ecore.resource.Resource;
 import org.eclipse.emf.transaction.RunnableWithResult;
 import org.eclipse.emf.transaction.TransactionalEditingDomain;
 import org.eclipse.emf.transaction.util.TransactionUtil;
 import org.eclipse.gmf.runtime.common.core.command.CommandResult;
 import org.eclipse.gmf.runtime.emf.commands.core.command.AbstractTransactionalCommand;
-import org.eclipse.xtext.ui.editor.validation.IValidationIssueProcessor;
 import org.eclipse.xtext.util.CancelIndicator;
 import org.eclipse.xtext.validation.CheckMode;
 import org.eclipse.xtext.validation.IResourceValidator;
 import org.eclipse.xtext.validation.Issue;
 import org.yakindu.sct.model.sgraph.resource.AbstractSCTResource;
 import org.yakindu.sct.ui.editor.DiagramActivator;
 
-import com.google.inject.Inject;
-
 /**
+ * The default implementation executes the validation within a readonly
+ * transaction on the transactional editing domain. during validation the model
+ * is locked and can not be changed.
  * 
  * @author andreas muelder - Initial contribution and API
  * 
  */
-public class SCTValidationJob extends Job {
-
-	@Inject
-	private IResourceValidator validator;
-
-	private IValidationIssueProcessor validationIssueProcessor;
+public class DefaultValidationJob extends ValidationJob {

```

Comment:
is relink really needed? see org.yakindu.sct.model.sgraph.resource.AbstractSCTResource.attachedHelper(EObject)